### PR TITLE
[cli] remove multiple servers support

### PR DESCRIPTION
**Pull request by bukepo**

### CODE REVIEW

File: Android.mk [hunk 1]
```diff
@@ -139,7 +139,6 @@ LOCAL_EXPORT_C_INCLUDE_DIRS     := \
 LOCAL_CPPFLAGS                                                              := \
     -std=c++11                                                                 \
     -pedantic-errors                                                           \
-    -Wno-non-virtual-dtor                                                      \
```
Comment (simonlingoogle) [inline Android.mk:142]: Is it related?

File: src/cli/cli.cpp [hunk 2]
```diff
@@ -361,29 +359,29 @@ int Interpreter::Hex2Bin(const char *aHex, uint8_t *aBin, uint16_t aBinLength, b
     return rval;
 }
 
-void Interpreter::AppendResult(otError aError) const
+void Interpreter::AppendResult(otError aError)
```
Comment (simonlingoogle) [inline src/cli/cli.cpp:362]: Can it be still `const`?

File: src/cli/cli.hpp [hunk 3]
```diff
@@ -183,7 +227,7 @@ class Interpreter
      * @retval  -1  Driver is broken.
      *
      */
-    int OutputIp6Address(const otIp6Address &aAddress) const;
+    int OutputIp6Address(const otIp6Address &aAddress);
```
Comment (simonlingoogle) [inline src/cli/cli.hpp:230]: Is it intentional to remove `const`?

File: src/cli/cli_uart.hpp [hunk 4]
```diff
@@ -48,16 +47,10 @@ namespace Cli {
  * This class implements the CLI server on top of the UART platform abstraction.
```
Comment (simonlingoogle) [inline src/cli/cli_uart.hpp:47]: revise comments too?

File: script/check-posix-pty [hunk 5]
```diff
@@ -139,7 +139,7 @@ EOF
 
     netstat -an | grep -q 61631 || die 'TMF port is not available!'
 
-    extaddr=$(awk '/extaddr/{getline; print}' $OT_OUTPUT | tr -d '\r\n')
+    extaddr=$(grep -aoE '[0-9a-z]{16}' $OT_OUTPUT)
```
Comment (simonlingoogle) [inline script/check-posix-pty:142]: Is it related?

File: src/cli/cli_config.h [hunk 6]
```diff
@@ -75,4 +77,17 @@
...
+#define OT_CLI_TRANSPORT_CONSOLE (2)
+
+/**
+ * @def OPENTHREAD_CONFIG_CLI_TRANSPORT
+ *
+ * The transport of the CLI.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_CLI_TRANSPORT
+#define OPENTHREAD_CONFIG_CLI_TRANSPORT OT_CLI_TRANSPORT_UART
```
Comment (simonlingoogle) [inline src/cli/cli_config.h:90]: Check `OPENTHREAD_CONFIG_CLI_TRANSPORT` must be 1 or 2?

File: Android.mk (hunk 1 above)
Comment (bukepo) [reply Android.mk:142]: yes, no virtual functions left.

File: src/cli/cli.cpp (hunk 2 above)
Comment (bukepo) [reply src/cli/cli.cpp:362]: no, the Output() function changes the internal state.

File: src/cli/cli.hpp (hunk 3 above)
Comment (bukepo) [reply src/cli/cli.hpp:230]: Yes, intentional.

File: script/check-posix-pty (hunk 5 above)
Comment (bukepo) [reply script/check-posix-pty:142]: Yes, without this change, this PR was easy to fail.

**Pull request merged**